pkg/metrics/controller_test.go

```diff
@@ -284,3 +285,36 @@ func TestControllerGameServersNodeState(t *testing.T) {
 		{labels: []string{"false"}, val: int64(2)},
 	})
 }
+
+func TestCalcDuration(t *testing.T) {
```

Comment:
Please, make this test to be a table-driven one and add a description for each test case like "what you test - what you expect". It will make this test much more readable, easy to understand, and easy to extend.
Example: https://github.com/googleforgames/agones/blob/master/pkg/apis/agones/v1/gameserver_test.go#L48